test/QueryTest.cc

```diff
@@ -1160,6 +1163,130 @@ TEST_CASE_METHOD(QueryTest, "Test Select All Result Key", "[Query]") {
     }
 }
 
+#pragma mark - ArrayIndex / Unnest
+
+// Spec: https://github.com/couchbaselabs/couchbase-lite-api/blob/master/spec/tests/T0004-Unnest-Array-Index.md
+// Version: 1.0.1
+//
+// Notes:
+// * 1. TestArrayIndexConfigInvalidExpressions is not applicable for CBL-C as
+//      ArrayIndexConfigurationConfiguration is just a pure struct (no validation logic)
+//
+
+/**
+ 2. TestCreateArrayIndexWithPath
+
+ Description
+ Test that creating an array index with only path works as expected.
+
+ Steps
+ 1. Load profiles.json into the collection named "_default.profiles".
+ 2. Create a ArrayIndexConfiguration object.
+     - path: "contacts"
+     - expressions:  null
+ 3. Create an array index named "contacts" in the profiles collection.
+ 4. Get index names from the profiles collection and check that the index named "contacts" exists.
+ 5. Get info of the index named "contacts" using an internal API and check that the index has
+     path and expressions as configured.
+ */
+TEST_CASE_METHOD(QueryTest, "TestCreateArrayIndexWithPath", "[Query]") {
+    auto collection = CreateCollection(db, "profiles");
+    ImportJSONLines("profiles_100.json", collection);
+    
+    CBLArrayIndexConfiguration config {};
+    config.path = "contacts"_sl;
+    
+    SECTION("SQL++") {
+        config.expressionLanguage = kCBLN1QLLanguage;
+    }
+    
+    SECTION("JSON") {
+        config.expressionLanguage = kCBLJSONLanguage;
+    }
```

Comment:
I haven't done this for iOS

Reply:
iOS only support N1QL language for the IndexConfiguration API.